### PR TITLE
[nanoleaf] Suppress ipv6 addr in controller discovery

### PR DESCRIPTION
**Pull request by stefan-hoehn**
The nanoleaf binding becomes very unstable when run in an ipv6 environment (though ipv6 is required when using matter). This simple change ignores the (additional) ipv6 announcement of the nanoleaf controller which fixes online/offline flapping of the nanoleaf thing.

Also see https://community.openhab.org/t/nanoleaf-binding-may-fail-when-upgrading-to-the-latest-nanoleaf-firmware-because-of-ipv6/148152

### CODE REVIEW

**Comment by stefan-hoehn** [comment]
@kaikreuzer Can you review this. It would make sense to add this the milestone tomorrow as matter is coming soon which requires ipv6 and without this fix the nanoleaf binding isn't reliable anymore.

**Comment by lsiepel** [comment]
Double check: does this apply to all nanoleave devices? All revisions and firmware?
So the devices have ipv6 but with ipv6 enabled all devices are unusable?

On my phone unable to fully asses the code change and don’t expect to have time before the milestone build

**Comment by stefan-hoehn** [comment]
Yes it does, it happens to all panels and canvases - so all "wall-mounted" devices (at least all that I have)

It is not related to bulbs and other devices that are only matter/thread enabled. 

If have many different devices of the ones that the binding supports and it is pretty annoying. The main problem seems somehow burried in the fact that the device may get communications via both ipv4 and ipv6 and therefore there seems to be something like an "interference" which then makes it flapping and it goes ON and OFF. As I am not completely sure what the exact issue is, this is, for the time being, the easiest solution that fixes it.  Maybe, in an upcoming release, I can figure it out and an idea would be that during discovery only one device creation is used, either ipv6 or ipv4 and as soon as this has been added the other will then be suppressed. However, this is a more tricky approach which I need to investigate. The only downside of this fix here is, that the binding doesn't support a device being used based on the ipv6 address which is probably not an issue.

**Comment by lsiepel** [comment]
Ok thanks, please add a note to the readme.md about this ipv6 suppression. Otherwise LGTM.

**Comment by stefan-hoehn** [comment]
I added a note to the Readme. 
Btw, it already hat a reference to that issue and mentioned "With firmware version 8.5.2 or newer..." IPv6 should be switched off which I have now removed as IPv6 can be kept on as the binding can deal with it during the discovery.

File: bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/discovery/NanoleafMDNSDiscoveryParticipant.java [hunk 1]
```diff
@@ -66,6 +68,15 @@ public String getServiceType() {
         }
         final Map<String, Object> properties = new HashMap<>(2);
         String host = service.getHostAddresses()[0];
+        try {
+            if (InetAddress.getByName(host).getAddress().length != 4) {
+                logger.debug("Ignoring IPv6 address for nanoleaf controllers: {}", host);
+                return null;
+            }
+        } catch (UnknownHostException e) {
+            logger.error("Error while checking IP address for nanoleaf controller: {}", host);
```
Comment (lsiepel) [inline bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/discovery/NanoleafMDNSDiscoveryParticipant.java:77]:
```suggestion
            logger.warn("Error while checking IP address for nanoleaf controller: {}", host);
```

**Review by lsiepel — CHANGES_REQUESTED** [review]
Otherwise LGTM

File: bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/discovery/NanoleafMDNSDiscoveryParticipant.java (hunk 1 above)
Comment (stefan-hoehn) [reply bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/discovery/NanoleafMDNSDiscoveryParticipant.java]: changed

**Review by lsiepel — APPROVED** [review]

**Pull request merged**